pkg/rbac/sharedData.go

```diff
@@ -100,6 +101,7 @@ func (shared *SharedData) GetClusterScopedResources(cache *Cache, ctx context.Co
 	defer shared.csLock.Unlock()
 	//clear previous cache
 	shared.csResources = nil
+	shared.csResourcesMap = make(map[Resource]struct{})
```

Comment:
Try to keep these consistent, use either `nil or make(...)` on both to avoid confusion.
Would be better if we can only use the map.